File: src/main/java/refactown/cleancode/c03functions/Fibonacci.java
package refactown.cleancode.c03functions;

public class Fibonacci {

	public static int fib(int x) {
		if (x == 0 || x == 1) {
			return x;
		}
		return fib(x - 1) + fib(x - 2);
	}

	public static String seq(int x) {
		if (x == 0) {
			return "0";
		}
		return "" + seq(x - 1) + "," + fib(x);
	}
}

File: src/main/kotlin/refactown/cleancode/c03functions/Fibonacci.kt
package refactown.cleancode.c03functions

fun fib(x: Int): Int = if (x in 0..1) x else fib(x - 1) + fib(x - 2)

fun seq(x: Int): String = if (x == 0) "0" else ""+seq(x-1) + "," + fib(x)